src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -2068,18 +2057,26 @@ private ExpressionEntry HandleCall(MethodDesc callee, MethodSignature signature,
             }
         }
 
-        private LLVMValueRef HandleCall(MethodDesc callee, MethodSignature signature, StackEntry[] argumentValues,
-            ILOpcode opcode, TypeDesc constrainedType, LLVMValueRef calliTarget, int offset, LLVMValueRef baseShadowStack, LLVMBuilderRef builder, bool needsReturnSlot,
-            LLVMValueRef castReturnAddress)
+        void AddDependencyForMethodCall(MethodDesc callee, ILOpcode opcode)
         {
             if (opcode == ILOpcode.callvirt && callee.IsVirtual)
             {
-                AddVirtualMethodReference(callee);
+                if (!callee.HasInstantiation)
+                {
+                    AddVirtualMethodReference(callee);
```

Comment:
I think the `AddVirtualMethodReference` calls became unnecessary after `WebAssemblyVTableSlotNode` got added. Maybe we can just delete them all?

The way these dependencies should show up in the system is implicitly through the `WebAssemblyVTableSlotNode` - it will inject the dependency for us.

Reply:
Sounds good, I deleted it.